lib/schema.js

```diff
@@ -13,18 +13,262 @@ class Schema {
+            name.includes('strange count transfer tool') ||
+            name.includes('strange bacon grease')
+        ) {
+            const schemaItem = this.getItemByItemName(name);
+            if (!schemaItem) return item;
+            item.defindex = schemaItem.defindex;
+            item.quality = item.quality || schemaItem.item_quality; //default quality
+            return item;
+        }
+
+        if (name.includes('strange')) {
+            name = name.replace('strange', '').trim();
+            item.quality = 11;
+        }
+        //Get all qualities
+        const qualities = Object.keys(this.raw.schema.qualities).reduce(
+            (obj, q) => {
+                obj[this.raw.schema.qualityNames[q].toLowerCase()] = this.raw.schema.qualities[q];
+                return obj;
+            },
+            {}
+        );
+        //Try to find quality name in name
+        for (const quality in qualities) {
+            if (name.includes(quality)) {
+                name = name.replace(quality, '').trim();
+                item.quality2 = item.quality;
+                item.quality = qualities[quality];
+                break;
+            }
+        }
+
+
+        const effects = this.raw.schema.attribute_controlled_attached_particles.reduce(
+            (obj, particle) => {
+                obj[particle.name.toLowerCase()] = particle.id;
+                return obj;
+            },
+            {}
+        );
+        //Check for effects
+        for (const effect in effects) {
+            if (name.includes(effect)) {
+                name = name.replace(effect, '').trim();
+                item.effect = effects[effect];
+                //will set quality to unusual if undefined, or make it primary, it other quality exists
+                if (item.quality != 5) {
+                    item.quality2 = item.quality;
+                    item.quality = 5;
+                }
+                break;
+            }
+        }
+        name = name.replace('uncraftable', 'non-craftable');
+        if (name.includes('non-craftable')) {
+            name = name.replace('non-craftable', '').trim();
+            item.craftable = false;
+        }
+        const wears = {
+            '(factory new)': 1,
+            '(minimal wear)': 2,
+            '(field-tested)': 3,
+            '(well-worn)': 4,
+            '(battle scarred)': 5
+        };
+        for (const wear in wears) {
+            if (name.includes(wear)) {
+                name = name.replace(wear, '').trim();
+                item.wear = wears[wear];
+                break;
+            }
+        }
+        const paintkits = Object.keys(this.raw.schema.paintkits).reduce((obj, id) => {
+            obj[this.raw.schema.paintkits[id].toLowerCase()] = +id;
+            return obj;
+        }, {});
+        for (const paintkit in paintkits) {
+            if (name.includes(paintkit)) {
+                name = name.replace(paintkit, '').trim();
```

Comment:
Need to add something here:
```js
    if (name.includes(paintkit) && name.includes(' | ')) {
        name = name.replace(paintkit, '').replace(' | ', '').trim();
        // ...
        }
```
![image](https://user-images.githubusercontent.com/47635037/111296803-dbb96780-8687-11eb-8e82-b944e928058e.png)
![image](https://user-images.githubusercontent.com/47635037/111296812-deb45800-8687-11eb-9bc3-394161c07645.png)

Reply:
Could you fix it ? I'm not sure if I can do so

Reply:
Sure thing. I will fix it later.